torchvision/prototype/datasets/utils/_internal.py

```diff
@@ -335,3 +340,47 @@ def _make_sharded_datapipe(root: str, dataset_size: int) -> IterDataPipe:
     # dp = dp.cycle(2)
     dp = TakerDataPipe(dp, dataset_size)
     return dp
+
+
+prod = functools.partial(functools.reduce, operator.mul)
+
+
+class NumericBinaryReader:
+    def __init__(self, file: IO, *, byte_order: str = sys.byteorder) -> None:
+        self._file = file
+        # torch.frombuffer interprets the bytes in the same byte order as the system. Thus, if the data is stored in
+        # the opposite byte order, we need to reverse the bytes before feeding them to torch.frombuffer().
+        self._reverse = byte_order != sys.byteorder
```

Comment:
Is there a way to let pytorch do that logic for us instead of having to manually take care of endianness? In numpy one can just add `<` or `>` to the dtype specification to indicate what is the expected encoding and everything gets taken care of

Reply:
Nope. See #4598 and #4651 for discussions about this.

Reply:
Then could we just rely on numpy for the reading part, since it's much more convenient (and possibly faster if one needs to account for endianness)?

Reply:
See https://github.com/pytorch/vision/pull/4882#issuecomment-963523041